File: week2/pset02/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
float calculate_average(float x, int y);

int main(void)
{
    string input = get_string("Text: ");

    int n_of_letters = count_letters(input);
    int n_of_words = count_words(input);
    int n_of_sentences = count_sentences(input);

    float L = calculate_average(n_of_letters, n_of_words);
    float S = calculate_average(n_of_sentences, n_of_words);

    double index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

int count_letters(string text)
{
    int count = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if (text[i] == 32)
        {
            count++;
        }
    }
    count++;
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    int n = strlen(text);
    for (int i = 0; i < n; i++)
    {
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            count++;
        }
    }
    return count;
}

float calculate_average(float x, int y)
{
    return x / y * 100;
}

File: week6/pset06/sentimental-readability/readability.py
from cs50 import get_string


def main():
    
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = calculate_average(letters, words)
    S = calculate_average(sentences, words)

    index = 0.0588 * L - 0.296 * S - 15.8

    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {int(round(index))}")


def count_letters(text):
    letter = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letter += 1
    return letter

def count_words(text):
    word = 0
    for i in range(len(text)):
        if " " in text[i]:
            word += 1
    word += 1
    return word

def count_sentences(text):
    sentence = 0
    for i in range(len(text)):
        if ("." in text[i]) or ("!" in text[i]) or ("?" in text[i]):
            sentence += 1
    return sentence


def calculate_average(x , y):
    return x / y * 100;


if __name__ == "__main__":
    main()